Extensions/Firebase/tests/FirebaseExtension.spec.js

```diff
@@ -0,0 +1,210 @@
+/**
+ * A firebase configuaration of a project made only for those tests.
+ */
+const firebaseConfig = {
+  apiKey: 'AIzaSyBwPnGpfEBXDjwQrWfU0wqgp4m9qEt7YM8',
+  authDomain: 'gdtest-e11a5.firebaseapp.com',
+  databaseURL: 'https://gdtest-e11a5.firebaseio.com',
+  projectId: 'gdtest-e11a5',
+  storageBucket: 'gdtest-e11a5.appspot.com',
+  messagingSenderId: '254035412678',
+  appId: '1:254035412678:web:2ddd6b83019b7f259b79c7',
+  measurementId: 'G-4REML26L59',
+};
+
+/**
+ * Turns a callback variable into a promise.
+ * @param {(callbackVariable: {setString: (result: "ok" | string) => void}) => any} executor
+ * @returns
+ */
+const promisifyCallbackVariables = (executor) =>
+  new Promise((done, err) => {
+    const callbackResult = new gdjs.Variable();
+    executor(
+      {
+        setString: (string) => {
+          if (string === 'ok') done(callbackResult);
+          else err(string);
+        },
+      },
+      callbackResult
+    );
+  });
+
+/** A complex variable using all variables types. */
+const variable = new gdjs.Variable();
+gdjs.evtTools.network._objectToVariable(
+  { my: 'test', object: { with: 'all', types: 2 }, it: ['is', true] },
+  variable
+);
+
+// The tests require an internet connection, as a real Firebase instance is used.
+const describeIfOnline = navigator.onLine ? describe : describe.skip;
+describeIfOnline('Firebase extension', function () {
```

Comment:
These are interesting tests but as you've identified they can be brittle/flaky in case we the network is done or something happens.
Usually, we avoid these tests, but they are still interesting "end to end" tests. So I suggest we:
- Mark them as such `Firebase extension end-to-end tests`.
- We can minimize the risk of concurrent executions breaking each other by using randomly generated field name when pushing and reading stuff in the database? 
  - This would at least ensure each test execution is independent.
  - There is still a slight risk of collision, but it's very very very tiny if we properly use a random name (and that would mean two tests run exactly at the same time... that's a one in several million even if we run the tests very often).
  - There is a risk of filling the database with test data, but we could also clean whatever is pushed to the database.